tests/test_samples.py

```diff
@@ -1339,3 +1339,140 @@ def test_old_gui():
         make_2d_axes(['x0', 'y0'], tex={'x0': '$x_0$', 'y0': '$y_0$'})
     with pytest.raises(NotImplementedError):
         make_1d_axes(['x0', 'y0'], tex={'x0': '$x_0$', 'y0': '$y_0$'})
+
+
+def test_groupby_stats():
+    mcmc = read_chains('./tests/example_data/cb')
+    chains = mcmc.groupby(('chain', '$n_\\mathrm{chain}$'))
```

Comment:
I think it might be better to set things up as follows:
```python
mcmc = read_chains('/home/lthergt/Documents/Projects/AnestheticPrj/anesthetic/tests/example_data/cb')
params = ['x0', 'x1']
chains = mcmc[params + ['chain']].groupby(('chain', '$n_\\mathrm{chain}$'))
```
and then in the following to only use `params` instead of all columns, since some columns are actually constant which can cause nans and warnings.

For example instead of `mcmc.loc[mask]` we would write `mcmc.loc[mask, params]` and get around of a lot of `'chain'` dropping.

This will also be closer to the actual use-cases.

Reply:
Agreed, and done in 706d759